### DISCUSSION TITLE
Module 1.1 RGB light not working - KeyError "Y" in try_experiment

### DISCUSSION THREAD

**Discussion by jp-lozano** [AccelerationConsortium/ac-microcourses#24] [Hello World] [answered]
The RGB light is not turning on at all, so the Demo continues to come back with an error. Any ideas as to why this is happening?

**Comment by sgbaird** [comment]
A few Q's:
1. Do you have a "waiting for experiment requests..." message on your microcontroller?
2. Did you run [4.2-paho-mqtt-colab-sdl-demo-test.ipynb](https://colab.research.google.com/github/sparks-baird/self-driving-lab-demo/blob/main/notebooks/4.2-paho-mqtt-colab-sdl-demo-test.ipynb) using the default HiveMQ credentials on both the microcontroller and the Colab notebook? (or ensure they are the same?)

**Reply by jp-lozano** [reply]
1. It says that before I run the program. It does run something (and the device beeps) but the actual LED does not turn on. Below I've included a screenshot of what i mean.
2. They should be, I have not made any changes to the HiveMQ credentials
<img width="610" alt="Screenshot 2024-07-17 at 13 04 21" src="https://github.com/user-attachments/assets/83d49f68-1400-47ae-8459-a1b72c380e52">

**Reply by sgbaird** [reply]
@jp-lozano which release version of the SDL Demo package are you using? Can you try downloading [the latest release](https://github.com/sparks-baird/self-driving-lab-demo/releases/latest) from scratch and reuploading the files? There's something strange with that "Y" appearing. I wonder if the file may have been accidentally changed and saved after you uploaded it. Also try closing out any open files in Thonny. There's an interesting quirk/feature to Thonny where a file open on the microcontroller has brackets around it e.g., `[main.py]`, whereas a file on your computer shows without the brackets in the tabname e.g., `main.py`.

**Reply by jp-lozano** [reply, marked as answer]
I had already tried this (and did check that i was opening files on the microcontroller). Disconnecting the grove to stemma QT ended up fixing the issue!

**Reply by sgbaird** [reply]
@jp-lozano the classic "unplug it and plug it in again" 😅 I will keep this in mind. Glad you got it figured out.
